tests/cloner_test.go

```diff
@@ -531,6 +531,73 @@ var _ = Describe("all clone tests", func() {
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+				//TODO:  get storageClass `cloneStorageClassName` and annotate with `cloneType`
+				storageProfile, err = f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+
+				Expect(err).ToNot(HaveOccurred())
+				storageProfile.Annotations["cdi.kubevirt.io/clone-strategy"] = string(cloneType)
+				_, err = f.CdiClient.CdiV1beta1().StorageProfiles().Update(context.TODO(), storageProfile, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(*storageProfile.Spec.CloneStrategy).To(Equal(cloneType))
+				Expect(*storageProfile.Status.CloneStrategy).To(Equal(cloneType))
+
+			})
+			AfterEach(func() {
+				By("[AfterEach] Restore the storage class - remove annotation ")
+				// TODO: Remove annotation from storageClass `cloneStorageClassName`
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), cloneStorageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				storageProfile.Annotations["cdi.kubevirt.io/clone-strategy"] = ""
+				_, err = f.CdiClient.CdiV1beta1().StorageProfiles().Update(context.TODO(), storageProfile, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(*storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(*storageProfile.Status.CloneStrategy).To(BeNil())
+
+				// TODO: verify that the storageProfile `cloneStorageClassName` has empty storageProfile.Status.CloneStrategy
+				storageProfile, err = f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).To(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).To(BeNil())
+
+			})
+			It("Should clone  with correct strategy from storageclass annotation ", func() {
+				//TODO: get storageProfile and verify taht the cloneStrategy is set to `cloneType`
+				storageProfile, err := f.CdiClient.CdiV1beta1().StorageProfiles().Get(context.TODO(), storageClassName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(storageProfile.Spec.CloneStrategy).ToNot(BeNil())
+				Expect(storageProfile.Status.CloneStrategy).ToNot(BeNil())
+				Expect(*storageProfile.Spec.CloneStrategy).To(Equal(cloneType))
+				Expect(*storageProfile.Status.CloneStrategy).To(Equal(cloneType))
+
+				pvcDef := utils.NewPVCDefinition(sourcePVCName, "1Gi", nil, nil)
+				pvcDef.Spec.StorageClassName = &cloneStorageClassName
+				pvcDef.Namespace = f.Namespace.Name
+
+				sourcePvc = f.CreateAndPopulateSourcePVC(pvcDef, sourcePodFillerName, fillCommand+testFile+"; chmod 660 "+testBaseDir+testFile)
+				doFileBasedCloneTest(f, pvcDef, f.Namespace, "target-dv", "1Gi")
```

Comment:
You are trying to clone before defining the targetDV. I don't think you need this line.

Reply:
done